[BUG]: `github_team_repository` error with version 5.9.0 - fails with `at least one permission expected from permissions map`

### Terraform Version

1.3.3

### Affected Resource(s)

- `github_team_repository`

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
resource "github_team_repository" "push_access" {
  for_each   = toset(var.push_access_team_ids)
  repository = github_repository.repository.id
  team_id    = each.value
  permission = "push"
}
```

### Debug Output
[PENDING]
Please provide a link to a GitHub Gist containing the complete debug output: https://www.terraform.io/docs/internals/debugging.html. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

### Panic Output
N/A

### Expected Behavior
Access should be refreshed, changes should be proposed by `terraform plan`

### Actual Behavior
`terraform plan` errors:

```shell
╷
│ Error: at least one permission expected from permissions map
│ 
│   with module.easyhotel_docker_compose_repository.github_team_repository.push_access["6845313"],
│   on modules/easyhotel_github_repository/main.tf line 34, in resource "github_team_repository" "push_access":
│   34: resource "github_team_repository" "push_access" {
│ 
╵
```

### Steps to Reproduce

#### With existing repo/permissions
* `terraform plan` will fail with above error

#### With non-existing permissions (permissions need to be created by this terraform run)
* `terraform apply` will create the permission/s above (verified that access was given by checking the GitHub web UI), but the apply will still fail with the same error above.

### Important Factoids
Is there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?

### References

Similar error messages (not the same cause I guess):
* https://stackoverflow.com/questions/51819294/terraform-error-at-least-one-permission-expected-from-permissions-map
* https://github.com/integrations/terraform-provider-github/issues/992

Same thing.

Terraform fails with API call, but the actual team's permission is added, I can see it in the gui.

Debug log:

```sh
✗  terraform version                                                                                  
Terraform v1.3.4
on linux_amd64


022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: 2022/11/18 21:51:39 [DEBUG] Github API Request Details:
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: ---[ REQUEST ]---------------------------------------
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: GET /organizations/11111111/team/2222222/repos/team-nest/team-backend HTTP/1.1
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Host: api.github.com
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: User-Agent: go-github/v48.0.0
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Accept: application/vnd.github.stone-crop-preview+json
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Accept-Encoding: gzip
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: 
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: 
2022-11-18T21:51:39.392-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: -----------------------------------------------------
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: 2022/11/18 21:51:39 [DEBUG] Github API Response Details:
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: ---[ RESPONSE ]--------------------------------------
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: HTTP/2.0 204 No Content
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Access-Control-Allow-Origin: *
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Access-Control-Expose-Headers: ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Content-Security-Policy: default-src 'none'
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Date: Sat, 19 Nov 2022 03:51:39 GMT
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Server: GitHub.com
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: Vary: Accept-Encoding, Accept, X-Requested-With
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Content-Type-Options: nosniff
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Frame-Options: deny
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Github-Media-Type: github.stone-crop-preview; format=json
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Github-Request-Id: 9BB6:09B3:4B1B1D:9AE311:637852C9
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Ratelimit-Limit: 12500
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Ratelimit-Remaining: 11289
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Ratelimit-Reset: 1668830354
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Ratelimit-Resource: core
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Ratelimit-Used: 1211
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: X-Xss-Protection: 0
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: 
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: 
2022-11-18T21:51:39.622-0600 [DEBUG] provider.terraform-provider-github_v5.9.0: -----------------------------------------------------
2022-11-18T21:51:39.622-0600 [ERROR] vertex "github_team_repository.backend" error: at least one permission expected from permissions map
2022-11-18T21:51:39.622-0600 [ERROR] vertex "github_team_repository.backend (expand)" error: at least one permission expected from permissions map
╷
│ Warning: Resource targeting is in effect
│ 
│ You are creating a plan with the -target option, which means that the result of this plan may not represent all of the changes requested by the current configuration.
│ 
│ The -target option is not for routine use, and is provided only for exceptional situations such as recovering from errors or mistakes, or when Terraform specifically suggests to use it as part of an error message.
╵
╷
│ Error: at least one permission expected from permissions map
│ 
│   with github_team_repository.backend,
│   on project-backend.tf line 77, in resource "github_team_repository" "backend":
│   77: resource "github_team_repository" "backend" {
│ 
╵
2022-11-18T21:51:39.851-0600 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2022-11-18T21:51:39.857-0600 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/integrations/github/5.9.0/linux_amd64/terraform-provider-github_v5.9.0 pid=2167833
2022-11-18T21:51:39.857-0600 [DEBUG] provider: plugin exited
make: *** [Makefile:54: apply] Error 1
```

terraform code creates repo, but then immediately fails with the error above

```hcl
resource "github_team_repository" "backend" {
  team_id    = module.github.github_team_id
  repository = module.github_repository_backend.id
  permission = "maintain"

  depends_on = [module.client, module.github, module.backend]
}
```

--- a/github/config.go
+++ b/github/config.go
@@ -6,6 +6,7 @@ import (
 	"net/http"
 	"net/url"
 	"path"
+	"strings"
 	"time"
 
 	"github.com/google/go-github/v48/github"
@@ -177,7 +178,7 @@ func newPreviewHeaderInjectorTransport(headers map[string]string, rt http.RoundT
 func (injector *previewHeaderInjectorTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	for name, value := range injector.previewHeaders {
 		header := req.Header.Get(name)
-		header += value
+		header = strings.Join([]string{header, value}, ",")
 		req.Header.Set(name, header)
 	}
 	return injector.rt.RoundTrip(req)

--- a/github/config.go
+++ b/github/config.go
@@ -39,6 +39,7 @@ func RateLimitedHTTPClient(client *http.Client, writeDelay time.Duration, readDe
 	client.Transport = NewRateLimitTransport(client.Transport, WithWriteDelay(writeDelay), WithReadDelay(readDelay))
 	client.Transport = logging.NewTransport("Github", client.Transport)
 	client.Transport = newPreviewHeaderInjectorTransport(map[string]string{
+		// TODO: remove when Stone Crop preview is moved to general availability in the GraphQL API
 		"Accept": "application/vnd.github.stone-crop-preview+json",
 	}, client.Transport)
 

--- a/github/config.go
+++ b/github/config.go
@@ -179,7 +179,11 @@ func newPreviewHeaderInjectorTransport(headers map[string]string, rt http.RoundT
 func (injector *previewHeaderInjectorTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	for name, value := range injector.previewHeaders {
 		header := req.Header.Get(name)
-		header = strings.Join([]string{header, value}, ",")
+		if header == "" {
+			header = value
+		} else {
+			header = strings.Join([]string{header, value}, ",")
+		}
 		req.Header.Set(name, header)
 	}
 	return injector.rt.RoundTrip(req)

--- a/github/config.go
+++ b/github/config.go
@@ -38,7 +38,6 @@ func RateLimitedHTTPClient(client *http.Client, writeDelay time.Duration, readDe
 	client.Transport = NewRateLimitTransport(client.Transport, WithWriteDelay(writeDelay), WithReadDelay(readDelay))
 	client.Transport = logging.NewTransport("Github", client.Transport)
 	client.Transport = newPreviewHeaderInjectorTransport(map[string]string{
-		// TODO: remove when Stone Crop preview is moved to general availability in the GraphQL API
 		"Accept": "application/vnd.github.stone-crop-preview+json",
 	}, client.Transport)
 
@@ -177,7 +176,9 @@ func newPreviewHeaderInjectorTransport(headers map[string]string, rt http.RoundT
 
 func (injector *previewHeaderInjectorTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	for name, value := range injector.previewHeaders {
-		req.Header.Set(name, value)
+		header := req.Header.Get(name)
+		header += value
+		req.Header.Set(name, header)
 	}
 	return injector.rt.RoundTrip(req)
 }

